contracts/tokens/MetaStudioToken.sol

```diff
@@ -97,21 +129,150 @@ contract MetaStudioToken is
     super._afterTokenTransfer(from, to, amount);
   }
 
+  /// @dev Using ERC777 secured implementation
   function _mint(address to, uint256 amount)
     internal
     override(ERC20Upgradeable, ERC20VotesUpgradeable)
   {
-    super._mint(to, amount);
+    _mint(to, amount, "", "");
   }
 
+  /// @dev Using ERC777 secured implementation
   function _burn(address account, uint256 amount)
     internal
     override(ERC20Upgradeable, ERC20VotesUpgradeable)
   {
-    super._burn(account, amount);
+    _burn(account, amount, "", "");
+  }
+
+  function canImplementInterfaceForAddress(
+    bytes32 interfaceHash,
+    address account
+  ) external view returns (bytes32) {
+    if (
+      account == address(this) &&
+      (interfaceHash == keccak256("ERC777Token") ||
+        interfaceHash == keccak256("ERC20Token"))
+    ) {
+      return "ERC1820_ACCEPT_MAGIC";
+    }
+    return "";
   }
 
+  /*
+   ERC777 overrided methodes-----
+   */
+
+  /// @inheritdoc ERC777Upgradeable
+  function approve(address spender, uint256 value)
+    public
+    override(ERC20Upgradeable, ERC777Upgradeable)
+    returns (bool)
+  {
+    return super.approve(spender, value);
+  }
+
+  function _approve(
+    address holder,
+    address spender,
+    uint256 value
+  ) internal override(ERC20Upgradeable, ERC777Upgradeable) {
+    super._approve(holder, spender, value);
+  }
+
+  function _spendAllowance(
```

Comment:
missing @inheritdoc ?

Reply:
Unused because internal function not exported to doc (.md file)